graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -542,15 +548,22 @@ def _rename_or_suppress_or_ignore_name_and_add_to_record(
         if (
             new_name_string in self.reverse_name_map
             and self.reverse_name_map[new_name_string] != name_string
-        ):
+        ) or new_name_string in self.custom_scalar_types:
+            # Either new_name_string is the name of a type (that may or may not have been
+            # renamed) in reverse_name_map or new_name_string is the name of a custom scalar.
+            # Either way, we want the original name of the type that is would be named
+            # new_name_string in the renamed schema.
+            other_name_string_renamed_to_new_name_string = self.reverse_name_map.get(
```

Comment:
not a big fan of this variable name

Reply:
The idea behind this error is when the original schema has two different types `Foo` and `Bar` that end up having the same name after calling `rename_schema`. In this case, `name_string` gets renamed to `new_name_string` when there's some other type (maybe a custom scalar type, maybe some other type) that we know will have the name `new_name_string` in the resulting schema, resulting in conflict. `other_name_string_renamed_to_new_name_string` is supposed to be the name of this other type.

Reply:
It seems like the `new_name_string` is the desired name for the type? If so, let's name it something like `desired_type_name`?

In general, I'm not a huge fan of the `_string` suffix or reverse Hungarian notation as a whole: https://en.wikipedia.org/wiki/Hungarian_notation

Hungarian notation is sometimes useful! For example: For example: https://www.joelonsoftware.com/2005/05/11/making-wrong-code-look-wrong/

These days, Hungarian notation's benefits can be replicated by `NewType()` in Python and similar tricks in other languages, and pure "this is a string" Hungarian (as opposed to "this int is a height, and not a width") is in general a code smell.

So I'd advise against using the `_string` suffix and instead focusing on the purpose of each variable in the algorithm you are constructing, then picking an appropriate name. Sometimes, `new` is good enough, or the best we can come up with. But frequently, we can do better.

In this case, `other_name_string_renamed_to_new_name_string` is simply `conflictingly_renamed_type_name`.